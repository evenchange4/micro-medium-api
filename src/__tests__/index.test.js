const micro = require('micro');
const listen = require('test-listen');
const request = require('request-promise');
const index = require('../index');

it('should run server without error', async () => {
  const service = micro(index);
  let error;
  try {
    const url = await listen(service);
    await request(url);
  } catch (err) {
    error = err;
  }

  expect(error).toBeUndefined();
  service.close();
});
