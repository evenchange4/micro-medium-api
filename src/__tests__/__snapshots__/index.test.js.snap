// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correct json data 1`] = `"{\\"data\\":{\\"posts\\":[{\\"title\\":\\"Microservice 產品交付\\",\\"firstPublishedAt\\":1502421324185,\\"url\\":\\"https://medium.com/@evenchange4/microservice-產品交付-9f2954c7167d\\",\\"content\\":{\\"subtitle\\":\\"Dockerize 與 Zeit JavaScript 跨平台解決方案\\"}},{\\"title\\":\\"使用 CircleCI 2.0 Workflows 挑戰三倍速\\",\\"firstPublishedAt\\":1501225933707,\\"url\\":\\"https://medium.com/@evenchange4/使用-circleci-2-0-workflows-挑戰三倍速-9691e54b0ef0\\",\\"content\\":{\\"subtitle\\":\\"減少 Monorepo CI 所需花費時間\\"}},{\\"title\\":\\"Centralized Error Handle in React with RxJS\\",\\"firstPublishedAt\\":1499912011732,\\"url\\":\\"https://medium.com/@evenchange4/centralized-error-handle-in-react-with-rxjs-fac9e9e37e44\\",\\"content\\":{\\"subtitle\\":\\"使用 Redux-Cylces 實行錯誤集中處理\\"}},{\\"title\\":\\"React Stack 開發體驗與優化策略\\",\\"firstPublishedAt\\":1499671540298,\\"url\\":\\"https://medium.com/@evenchange4/react-stack-開發體驗與優化策略-b056da2fa0aa\\",\\"content\\":{\\"subtitle\\":\\"Landing Page for MCS Lite\\"}},{\\"title\\":\\"Component-based Code Splitting\\",\\"firstPublishedAt\\":1495693600421,\\"url\\":\\"https://medium.com/@evenchange4/component-based-code-splitting-70aeebbb2dc7\\",\\"content\\":{\\"subtitle\\":\\"Lazy Load React Component with React-Loadable\\"}},{\\"title\\":\\"這個雷踩了四次，為什麼你可能需要 Eslint\\",\\"firstPublishedAt\\":1494422818494,\\"url\\":\\"https://medium.com/@evenchange4/這個雷踩了四次-為什麼你可能需要-eslint-9330c1245ec1\\",\\"content\\":{\\"subtitle\\":\\"以一個簡單到不行的大小寫為例，不單只是 Coding Style 或命名規範，你可能會需要 Eslint 來幫你盡量降低這種小意外的發生。\\"}},{\\"title\\":\\"Reproducing Medium Style Progressive Image Loading for React\\",\\"firstPublishedAt\\":1493908671942,\\"url\\":\\"https://medium.com/@evenchange4/reproducing-medium-style-progressive-image-loading-for-react-2e83bba0c608\\",\\"content\\":{\\"subtitle\\":\\"Recently, I have read an article by José M. Pérez about How Medium does progressive image loading, and that is what I need about the lazy…\\"}},{\\"title\\":\\"Build A Web App in MediaTek\\",\\"firstPublishedAt\\":1493193304597,\\"url\\":\\"https://medium.com/@evenchange4/build-a-web-app-in-mediatek-61b0a26215a0\\",\\"content\\":{\\"subtitle\\":\\"MCS Lite’s Front-End Tech Stack for Mobile\\"}},{\\"title\\":\\"Writing a Library for LinkIt 7697\\",\\"firstPublishedAt\\":1493083372209,\\"url\\":\\"https://medium.com/@evenchange4/writing-a-library-for-linkit-7697-9e562d28e17\\",\\"content\\":{\\"subtitle\\":\\"開發板の隨手筆記 - 封裝 MCS API for Arduino\\"}},{\\"title\\":\\"I18n Workflow for React Project\\",\\"firstPublishedAt\\":1492745328047,\\"url\\":\\"https://medium.com/@evenchange4/i18n-workflow-for-react-project-9f9ff8fe9aef\\",\\"content\\":{\\"subtitle\\":\\"多國語言的機制建議盡量在專案初期就能有好的規劃，不至中後期要把字串抽出將會是件累人的事，而在現今前端開發的演進下，多國語言的流程也出現了一些不同的處理方式，以下簡介我們在 MediaTek 內前端專案處理的模式。\\"}}],\\"user\\":{\\"username\\":\\"evenchange4\\",\\"name\\":\\"Michael Hsu\\",\\"bio\\":\\"Front-End Developer | https://michaelhsu.tw/\\"}}}"`;
